plugins/source/postgresql/client/sync.go

```diff
@@ -0,0 +1,131 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/cloudquery/plugin-sdk/plugins/source"
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/jackc/pgx/v5"
+)
+
+func (c *Client) Sync(ctx context.Context, metrics *source.Metrics, res chan<- *schema.Resource) error {
+	// var conn *pgconn.PgConn
+	var err error
+	var snapshotName string
+	c.metrics = metrics
+	for _, table := range c.Tables {
+		if c.metrics.TableClient[table.Name] == nil {
+			c.metrics.TableClient[table.Name] = make(map[string]*source.TableClientMetrics)
+			c.metrics.TableClient[table.Name][c.ID()] = &source.TableClientMetrics{}
+		}
+	}
+
+	connPool, err := c.Conn.Acquire(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to acquire connection: %w", err)
+	}
+	// this must be closed only at the end of the initial sync process otherwise the snapshot
+	// used to sync the initial data will be released
+	defer connPool.Release()
+	conn := connPool.Conn().PgConn()
+
+	if c.pluginSpec.CDC {
+		snapshotName, err = c.startCDC(ctx, conn)
+		if err != nil {
+			return err
+		}
+	}
+
+	if c.pluginSpec.CDC && snapshotName == "" {
```

Comment:
What happens if someone kills (e.g. ctrl+c) the CLI during the initial sync? Won't that result in missing data since next time we'll skip it? Another use case is a batch failing to write due to a bug in the destination. We fix the bug in the destination and re-sync. Will that give us the new data?

Maybe always do the initial sync?

Reply:
If we kill in the initial sync we can really recover as snapshots in postgres are only hold per transaction. So this is also not something supported in `pgdump` as there is no feature in Postgres that can export snapshots outside of a transaction. 

So if it stop in the middle the best thing we can potentially do is delete the replication slot.